File: integration/typescript/examples/build_mint.ts
import {
    buildMintDirectInstruction,
    loadMintDirectOperation,
    serializeInstruction,
} from "../src/index.ts";

const operationFile = process.argv[2];
if (operationFile === undefined) {
    throw new Error("Usage: build_mint.ts <direct-mint.operation.json>");
}

const operation = loadMintDirectOperation(operationFile);
const instruction = buildMintDirectInstruction(operation);
process.stdout.write(serializeInstruction(instruction));

File: rust/examples/build_mint.rs
use std::{env, error::Error};

use chancery_reference::{
    IntegrationError,
    build_mint_direct_instruction,
    load_mint_direct_operation,
    mint_instruction_to_document,
};

fn main() -> Result<(), Box<dyn Error>> {
    let operation_file = env::args_os().nth(1).ok_or_else(|| {
        IntegrationError::new("Usage: build_mint <direct-mint.operation.json>")
    })?;
    let (accounts, arguments) = load_mint_direct_operation(operation_file)?;
    let instruction = build_mint_direct_instruction(&accounts, &arguments)?;
    let document = mint_instruction_to_document(&instruction)?;
    println!("{}", serde_json::to_string_pretty(&document)?);
    Ok(())
}
